[WIP] Add Manifest Stats in snapshot summary.

nk1506 opened the pull request.

nk1506 committed:
Add Manifest Stats in snapshot summary.

ajantha-bhat commented on core/src/main/java/org/apache/iceberg/SnapshotSummary.java:
```diff
@@ -336,6 +352,16 @@ void addedManifest(ManifestFile manifest) {
+    void addedManifestStats(ManifestFile manifest) {
+      switch (manifest.content()) {
+        case DATA:
+          this.totalDataManifestFiles++;
```
why not increment these in `addedManifest` method itself? why we need a separate method?

nk1506 replied:
It seems addedManifest is being used with Spark import action only. 
Ref: https://github.com/apache/iceberg/blob/01bc864b8eb8c4ca4240af85f00f4e67c78c466e/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java#L573

ajantha-bhat commented on core/src/test/java/org/apache/iceberg/TestSnapshot.java:
```diff
@@ -275,4 +287,22 @@ private void runAddedDeleteFileSequenceNumberTest(
+                .count();
+    assertThat(table.currentSnapshot().summary().get(TOTAL_DATA_MANIFEST_FILES))
+        .isEqualTo(String.valueOf(dataManifestCount));
+    int deletedManifestCount =
```
```suggestion
    int deleteManifestCount =
```

ajantha-bhat commented on core/src/test/java/org/apache/iceberg/TestSnapshot.java:
```diff
@@ -275,4 +287,22 @@ private void runAddedDeleteFileSequenceNumberTest(
   }
+
+  public static void testManifestStats(Table table) {
+    List<ManifestFile> manifestFiles = table.currentSnapshot().allManifests(table.io());
```
since it is a test case, if we use, currentSnapshot().dataManifests and currentSnapshot().deleteManifests, it can look simpler.

ajantha-bhat replied:
Also, allManifests is filling the `dataManifests` and `deleteManifests` fields if it is null. So, with one IO we can get both the fields.

ajantha-bhat commented on spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java:
```diff
@@ -201,7 +203,7 @@ public void testBinPackPartitionedTable() {
     shouldHaveFiles(table, 4);
     List<Object[]> actualRecords = currentData();
-
+    testManifestStats(table);
```
nit: it should be placed before computing `actualRecords`, check all the places. 

Also, keep the new line as before for all the places.

tomtongue committed:
Flink: Remove MiniClusterResource (#10817)

lurnagao-dahua committed:
Docs: Use link addresses instead of descriptions in releases.md (#10815)

devinrsmith committed:
Build: Declare avro as an api dependency of iceberg-core (#10573)

iceberg-core should declare an api dependency on avro. For example, the public class `org.apache.iceberg.PartitionData` extends avro-specific types. In addition, there are public methods that deal with avro types, see `org.apache.iceberg.avro.AvroSchemaUtil`

stevenzwu committed:
Flink: backport PR #10748 for limit pushdown (#10813)

Commit:
Docs: Fix header for entries metadata table (#10826)

huaxingao committed:
Spark 3.5: Support Reporting Column Stats (#10659)


Co-authored-by: Karuppayya Rajendran <karuppayya1990@gmail.com>

venkata91 committed:
Flink: Backport #10548 to v1.18 and v1.17 (#10776)

nastra committed:
Infra: Improve feature request template (#10825)

hsiang-c committed:
Core: Replace the duplicated ALL_DATA_FILES with ALL_DELETE_FILES (#10836)

RussellSpitzer committed:
Core: Adds Basic Classes for Iceberg Table Version 3 (#10760)

grantatspothero committed:
Core: Allow SnapshotProducer to skip uncommitted manifest cleanup after commit (#10523)

stevenzwu committed:
Flink: a few small fixes or tuning for range partitioner (#10823)

findepi committed:
Drop support for Java 8 (#10518)

nastra committed:
Build: Bump com.adobe.testing:s3mock-junit5 from 2.11.0 to 2.17.0 (#10851)

nastra committed:
Core: Upgrade Jetty and Servlet API (#10850)

This is the latest Jetty version that runs with JDK11

snazy committed:
Build: Configure options.release = 11 / remove com.palantir.baseline-release-compatibility plugin  (#10849)

dependabot[bot] committed:
Build: Bump kafka from 3.7.1 to 3.8.0 (#10797)

Bumps `kafka` from 3.7.1 to 3.8.0.

Updates `org.apache.kafka:kafka-clients` from 3.7.1 to 3.8.0

Updates `org.apache.kafka:connect-api` from 3.7.1 to 3.8.0

Updates `org.apache.kafka:connect-json` from 3.7.1 to 3.8.0

---
updated-dependencies:
- dependency-name: org.apache.kafka:kafka-clients
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: org.apache.kafka:connect-api
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: org.apache.kafka:connect-json
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

findepi committed:
Build: Update baseline gradle plugin to 5.58.0 (#10788)

stevenzwu committed:
Flink: refactor sink tests to reduce the number of combinations with parameterized tests (#10777)

stevenzwu committed:
Flink: backport PR #10823 for range partitioner fixup (#10847)

findepi committed:
Core: Remove reflection from TestParallelIterable (#10857)

This is a unit test, so can leverage package-private access.

rdblue committed:
Spec: Deprecate the file system table scheme (#10833)

shanielh committed:
Core, API: UpdatePartitionSpec: Added ability to create a new Partition Spec but not set it as the Default

dependabot[bot] committed:
Build: Bump com.palantir.baseline:gradle-baseline-java (#10864)

Bumps [com.palantir.baseline:gradle-baseline-java](https://github.com/palantir/gradle-baseline) from 5.58.0 to 5.61.0.
- [Release notes](https://github.com/palantir/gradle-baseline/releases)
- [Changelog](https://github.com/palantir/gradle-baseline/blob/develop/.changelog.yml)
- [Commits](https://github.com/palantir/gradle-baseline/compare/5.58.0...5.61.0)

---
updated-dependencies:
- dependency-name: com.palantir.baseline:gradle-baseline-java
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build: Bump nessie from 0.94.2 to 0.94.4 (#10869)

Bumps `nessie` from 0.94.2 to 0.94.4.

Updates `org.projectnessie.nessie:nessie-client` from 0.94.2 to 0.94.4

Updates `org.projectnessie.nessie:nessie-jaxrs-testextension` from 0.94.2 to 0.94.4

Updates `org.projectnessie.nessie:nessie-versioned-storage-inmemory-tests` from 0.94.2 to 0.94.4

Updates `org.projectnessie.nessie:nessie-versioned-storage-testextension` from 0.94.2 to 0.94.4

---
updated-dependencies:
- dependency-name: org.projectnessie.nessie:nessie-client
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.projectnessie.nessie:nessie-jaxrs-testextension
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.projectnessie.nessie:nessie-versioned-storage-inmemory-tests
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.projectnessie.nessie:nessie-versioned-storage-testextension
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build: Bump org.xerial:sqlite-jdbc from 3.46.0.0 to 3.46.0.1 (#10871)

Bumps [org.xerial:sqlite-jdbc](https://github.com/xerial/sqlite-jdbc) from 3.46.0.0 to 3.46.0.1.
- [Release notes](https://github.com/xerial/sqlite-jdbc/releases)
- [Changelog](https://github.com/xerial/sqlite-jdbc/blob/master/CHANGELOG)
- [Commits](https://github.com/xerial/sqlite-jdbc/compare/3.46.0.0...3.46.0.1)

---
updated-dependencies:
- dependency-name: org.xerial:sqlite-jdbc
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build: Bump org.apache.commons:commons-compress from 1.26.0 to 1.26.2 (#10868)

Bumps org.apache.commons:commons-compress from 1.26.0 to 1.26.2.

---
updated-dependencies:
- dependency-name: org.apache.commons:commons-compress
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build: Bump software.amazon.awssdk:bom from 2.26.25 to 2.26.29 (#10866)

Bumps software.amazon.awssdk:bom from 2.26.25 to 2.26.29.

---
updated-dependencies:
- dependency-name: software.amazon.awssdk:bom
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build: Bump mkdocs-material from 9.5.30 to 9.5.31 (#10863)

Bumps [mkdocs-material](https://github.com/squidfunk/mkdocs-material) from 9.5.30 to 9.5.31.
- [Release notes](https://github.com/squidfunk/mkdocs-material/releases)
- [Changelog](https://github.com/squidfunk/mkdocs-material/blob/master/CHANGELOG)
- [Commits](https://github.com/squidfunk/mkdocs-material/compare/9.5.30...9.5.31)

---
updated-dependencies:
- dependency-name: mkdocs-material
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

snazy committed:
Build: Fix Scala compilation (#10860)

`ScalaCompile` does not respect `options.release` and `-release:11` in the aruments is not enough. Re-adding `sourceCompatibility` + `targetCompatibility` for 11.

findepi committed:
Build: Enable FormatStringAnnotation error-prone check (#10856)

nastra committed:
Core: Use encoding/decoding methods for namespaces and deprecate Splitter/Joiner (#10858)

stevenzwu committed:
Flink: move v1.19 to v.120

stevenzwu committed:
Flink: remove v1.17 module

stevenzwu committed:
Flink: add v1.19 back after coping from 1.20

stevenzwu committed:
Flink: adjust code for the new 1.20 module.

also fixed the bug of missing jmh in the 1.19 module.

nastra committed:
Aliyun: Replace assert usage with assertThat (#10880)

deniskuzZ committed:
Core: Extract filePath comparator into it's own class (#10664)

nakaken-churadata committed:
Docs: Fix SQL in branching docs (#10876)

amogh-jahagirdar committed:
API: Add SupportsRecoveryOperations mixin for FileIO (#10711)

emkornfield committed:
Spec: Clarify identity partition edge cases (#10835)

dependabot[bot] committed:
Build: Bump org.testcontainers:testcontainers from 1.20.0 to 1.20.1 (#10865)

Bumps [org.testcontainers:testcontainers](https://github.com/testcontainers/testcontainers-java) from 1.20.0 to 1.20.1.
- [Release notes](https://github.com/testcontainers/testcontainers-java/releases)
- [Changelog](https://github.com/testcontainers/testcontainers-java/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testcontainers/testcontainers-java/compare/1.20.0...1.20.1)

---
updated-dependencies:
- dependency-name: org.testcontainers:testcontainers
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nastra committed:
Build: Add checkstyle rule to ban assert usage (#10886)

Fokko committed:
Build: Bump Apache Avro to 1.12.0 (#10879)

findepi committed:
Spec: Fix rendering of unified partition struct (#10896)

The angle brackets were without any escapes so docs renderer treated
them as HTML. The resulting text on the website looked like an
unfinished sentence:

    The unified partition type looks like Struct.

Putting the angle brackets in backticks prevent them from being
interpreted as HTML. Surrounding names like spec#0, field#1 are also put
inside backticks for consistence.

tomtongue committed:
Docs: Fix catalog name for S3 MRAP example (#10897)

snazy committed:
Add Flink 1.20 & remove Flink 1.17 in stage-binaries.sh and docs (#10888)

This is a follow-up to #10881

findepi committed:
Flink: Remove deprecated RowDataUtil.clone method (#10902)

Scheduled for removal in 1.7.0.

nastra committed:
AWS: Fix flaky TestS3RestSigner (#10898)

amogh-jahagirdar committed:
AWS: Implement SupportsRecoveryOperations mixin for S3FileIO (#10721)

nk1506 committed:
Core: Remove deprecated APIs for 1.7.0 (#10818)

nk1506 committed:
Core, Flink: Fix build warnings (#10899)

manuzhang committed:
Build: Bump Spark 3.5 to 3.5.2 (#10918)

nk1506 committed:
Fixed conflicts
